### PR TITLE
Added the unique API to crypto utils

### PR DESCRIPTION
**Pull request by Randynamic**
## Reason:
Moved implementation from `kadena.js` package to `cryptography-utils`

### CODE REVIEW

**Review by ggobugi27 — APPROVED** [review]

**Comment by ggobugi27** [comment]
Looks like the imports failed in `createListenRequest`, and other `kadena.js/api` functions

File: packages/libs/cryptography-utils/src/unique.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+export function unique(hashes: Array<IBase64Url>): Array<IBase64Url> {
+  const isUnique: IHashes = {};
+  return hashes.filter((hash) => {
+    if (!isUnique[hash]) {
+      isUnique[hash] = true;
+      return true;
+    }
+    return false;
+  });
+}
```
Comment (alber70g) [inline packages/libs/cryptography-utils/src/unique.ts:38]: Wouldn't it be better to leave filtering doubles to the consuming library?

Comment (Randynamic) [reply packages/libs/cryptography-utils/src/unique.ts]: I agree as this should only be a helper function.
But we first need to comply with the previous `pact-lang-api` API before deprecating or changing the API

File: packages/libs/cryptography-utils/src/unique.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+export function unique(array: Array<string>): Array<string> {
+  const isUnique: ICache = {};
+  return array.filter((item: string) => {
+    if (!isUnique[item]) {
+      isUnique[item] = true;
+      return true;
+    }
+    return false;
+  });
+}
```
Comment (alber70g) [inline packages/libs/cryptography-utils/src/unique.ts:36]: Maybe we can make this generic?
```suggestion
export function unique<T extends string>(hashes: Array<T>): Array<T> {
  const isUnique: Set<T> = {} as Set<T>;
   hashes.forEach((hash) => {
    if (!isUnique.has(hash)) {
      isUnique.add(hash);
    }
  });
  return Array.from(isUnique)
}
```

File: packages/libs/cryptography-utils/src/unique.ts (hunk 1 above)
Comment (alber70g) [reply packages/libs/cryptography-utils/src/unique.ts]: Can we add the [`@deprecated`](https://api-extractor.com/pages/tsdoc/tag_deprecated/) tag?

**Review by ggobugi27 — APPROVED** [review]

**Pull request merged**